sycl/doc/extensions/DeviceGlobal/SYCL_INTEL_device_global.asciidoc

```diff
@@ -0,0 +1,790 @@
+
+A sketch of the anticipated constructor interface is:
+
+[source,c++]
+----
+namespace sycl::ext::oneapi {
+
+template <typename T, typename propertyListT = property_list<>>
+class device_global {
+public:
+  using element_type = std::remove_extent_t<T>; 
+  
+  static_assert(std::is_trivially_destructible_v<T>,
+      "Type T must be trivially destructible.");
+
+  #ifdef __cpp_consteval
+    device_global();
+
+    // device_global initializes underlying T with the args argument
+    template <typename... Args>
+    consteval explicit device_global(Args&&... args);
+  #else
+    static_assert(std::is_trivially_default_constructible_v<T>,
+                  "Type T must be trivially default constructable (until C++20 "
+                  "consteval is supported and enabled)");
+
+    device_global();
+  #endif // __cpp_consteval
+
+----
+
+The example below creates two global namespace scope `device_global` objects named `dm1` and `dm2`. `dm1` is default constructed with external linkage, while `dm2` is initialized and has internal linkage.
+
+[source,c++]
+----
+using namespace sycl;
+using namespace sycl::ext::oneapi;
+
+device_global<MyClass> dm1;
+static device_global<int[4]> dm2{1, 3, 5, 7};  // Requires C++20 to be enabled
+
+int main () {
+  sycl::queue Q;
+  Q.submit([&](sycl::handler& h) {
+    h.single_task([=]() {
+      int x = 5;
+      if (dm1.get().flag)
+        x = dm2[0];
+    });
+  });
+}
+----
+
+== Non-normative: Implementation hints
+
+`device_global` prioritizes usability over simplicity of implementation, and therefore adds requirements such as (1) that contents and addresses of the allocation on each device remain stable across changes to specialization constant values, and (2) that the allocation be accessible across `device_image` on the same device.  These requirements mean that the semantics of `device_global` do not match the semantics of SPIR-V module scope variables, and therefore may not be implementable exclusively using the SPIR-V feature in existing SPIR-V consuming implementations.
+
+Also note that there are no restrictions on passing (and subsequent dereferencing) of pointers obtained on a device from a `device_global`, between kernels on a device, including through storage to memory.
+
+== Issues
+
+1) Can `sycl::atomic_ref` be used with `device_global`? +
+*Resolved*: Yes, but only on the device side.  There is no visibility/communication across devices because each device receives a unique allocation of type _T_ underlying the `device_global`.  There is no way for an `atomic_ref` to the to be created in host code because there is no way to extract a pointer or reference in host code (only copy/memcpy).
```

Comment:
```suggestion
*Resolved*: Yes, but only on the device side.  There is no visibility/communication across devices because each device receives a unique allocation of type _T_ underlying the `device_global`.  There is no way for an `atomic_ref` associated with the allocation to be created in host code because there is no way to extract a pointer or reference in host code (only copy/memcpy).
```